From 72123e1b56f53f9205bb105f8a62d0869b837b22 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Wed, 26 Jan 2022 20:20:43 -0800
Subject: [PATCH 93/93] NEWS: Add a bug entry for BZ #28755

---
 NEWS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/NEWS b/NEWS
index c063c78b15..daeb066419 100644
--- a/NEWS
+++ b/NEWS
@@ -50,6 +50,7 @@ The following bugs are resolved with this release:
   [28704] elf/tst-cpu-features-cpuinfo fails for KVM guests on some AMD systems
   [28707] assert in tzfile.c __tzfile_read striking with truncated timezones
   [28744] A64FX string functions are selected without SVE HWCAP
+  [28755] overflow bug in wcsncmp_avx2 and wcsncmp_evex
   [28771] %ebx optimization macros are incompatible with .altmacro
   [28768] CVE-2022-23218: Buffer overflow in sunrpc svcunix_create
   [28769] CVE-2021-3999: Off-by-one buffer overflow/underflow in getcwd()
-- 
2.35.0

