From 99978cd42c55ee427fb512de69638045f6d525c7 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Wed, 8 Jun 2022 09:40:28 -0300
Subject: [PATCH 130/134] NEWS: Add a bug fix entry for BZ #29225

---
 NEWS | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/NEWS b/NEWS
index 91acc73df7..8d13863133 100644
--- a/NEWS
+++ b/NEWS
@@ -41,6 +41,8 @@ The following bugs are resolved with this release:
   [29211] libc: __open_catalog is not y2038 aware
   [29213] libc: gconv_parseconfdir is not y2038 aware
   [29214] nptl: pthread_setcanceltype fails to set type
+  [29225] network: Mistyped define statement in socket/sys/socket.h in
+    line 184
 
 
 Version 2.35
-- 
2.36.1

