diff -up glibc-2.34/rt/Makefile.39~ glibc-2.34/rt/Makefile
--- glibc-2.34/rt/Makefile.39~	2021-08-02 10:35:18.780415863 +0200
+++ glibc-2.34/rt/Makefile	2021-08-02 10:35:56.864655876 +0200
@@ -114,3 +114,13 @@ endif
 endif # !$(pthread-in-libc)
 
 tst-mqueue7-ARGS = -- $(host-test-program-cmd)
+
+# XXX avoid timing issues on fine-grained SMT systems
+ifeq (powerpc, $(base-machine))
+no-parallel-testing = yes
+endif
+ifneq ($(filter %tests,$(MAKECMDGOALS)),)
+ifeq ($(no-parallel-testing),yes)
+.NOTPARALLEL:
+endif
+endif
