--- glibc-2.16.90-54a41734/crypt/crypt-entry.c.orig	2012-10-31 17:45:54.367377046 -0200
+++ glibc-2.16.90-54a41734/crypt/crypt-entry.c	2012-10-31 17:46:46.605380843 -0200
@@ -183,18 +183,3 @@ crypt (key, salt)
 }
 #endif
 
-/*
- * To make fcrypt users happy.
- * They don't need to call init_des.
- */
-#ifdef _LIBC
-weak_alias (crypt, fcrypt)
-#else
-char *
-__fcrypt (key, salt)
-     const char *key;
-     const char *salt;
-{
-  return crypt (key, salt);
-}
-#endif
--- glibc-2.16.90-54a41734/crypt/wrapper.c.orig	2012-10-31 17:47:03.054380797 -0200
+++ glibc-2.16.90-54a41734/crypt/wrapper.c	2012-10-31 17:47:14.397381034 -0200
@@ -324,7 +324,6 @@ weak_alias(__crypt, crypt)
 weak_alias(__crypt_gensalt_rn, crypt_gensalt_rn)
 weak_alias(__crypt_gensalt_ra, crypt_gensalt_ra)
 weak_alias(__crypt_gensalt, crypt_gensalt)
-weak_alias(crypt, fcrypt)
 #endif
 
 #ifdef TEST
