--- libc/stdlib/Makefile.ldbl~	2012-11-06 18:31:45.000000000 +0100
+++ libc/stdlib/Makefile	2012-12-27 20:03:07.397489350 +0100
@@ -141,9 +141,15 @@ tests: $(objpfx)tst-fmtmsg.out
 endif
 endif
 
+isomac-includes = -I../include $(+sysdep-includes) $(sysincludes) -I..
+ifeq (powerpc:8, $(base-machine):$(sizeof-long-double))
+# XXX only necessary for bootstrapping...
+isomac-includes += -I../libio -I../stdlib -I../wcsmbs
+endif
 $(objpfx)isomac.out: $(objpfx)isomac
 	$(dir $<)$(notdir $<) '$(CC)'  \
 	'-I../include $(+sysdep-includes) $(sysincludes) -I..' > $<.out
+	'$(isomac-includes)' > $<.out
 
 isomac-CFLAGS = -O
 $(objpfx)isomac: isomac.c
