--- libc/elf/setup-vdso.h.bero	2012-12-27 19:36:02.133987793 +0100
+++ libc/elf/setup-vdso.h	2012-12-27 19:36:36.764572111 +0100
@@ -93,7 +93,9 @@ setup_vdso (struct link_map *main_map __
 	  char *copy = malloc (len);
 	  if (copy == NULL)
 	    _dl_fatal_printf ("out of memory\n");
-	  l->l_libname->name = l->l_name = memcpy (copy, dsoname, len);
+	  l->l_libname->name = memcpy (copy, dsoname, len);
+          if (GLRO(dl_debug_mask))
+            l->l_name = copy;
 	}
 
       /* Add the vDSO to the object list.  */
