--- glibc-2.23/aclocal.m4.shareLocale~	2016-02-18 20:59:26.420203855 +0100
+++ glibc-2.23/aclocal.m4	2016-02-18 21:01:00.898654328 +0100
@@ -296,7 +296,7 @@ case "$prefix" in
   if test "$libdir" = '${exec_prefix}/lib'; then
     libdir='${exec_prefix}/$1';
     # Locale data can be shared between 32-bit and 64-bit libraries.
-    libc_cv_complocaledir='${exec_prefix}/lib/locale'
+    libc_cv_complocaledir='${exec_prefix}/share/locale'
   fi
   ;;
 esac])
--- glibc-2.23/Makeconfig.shareLocale~	2016-02-18 20:59:26.421203860 +0100
+++ glibc-2.23/Makeconfig	2016-02-18 21:01:25.644772357 +0100
@@ -193,7 +193,7 @@ inst_zonedir = $(install_root)$(zonedir)
 # Where to install the compiled binary locale archive and compiled
 # binary locale files.
 ifndef complocaledir
-complocaledir = $(libdir)/locale
+complocaledir = $(datadir)/locale
 endif
 inst_complocaledir = $(install_root)$(complocaledir)
 
--- glibc-2.23/sysdeps/gnu/configure.shareLocale~	2016-02-18 20:59:26.421203860 +0100
+++ glibc-2.23/sysdeps/gnu/configure	2016-02-18 21:01:45.136865335 +0100
@@ -14,7 +14,7 @@ case "$prefix" in
   if test "$libdir" = '${exec_prefix}/lib'; then
     libdir='${exec_prefix}/lib';
     # Locale data can be shared between 32-bit and 64-bit libraries.
-    libc_cv_complocaledir='${exec_prefix}/lib/locale'
+    libc_cv_complocaledir='${exec_prefix}/share/locale'
   fi
   ;;
 esac
--- glibc-2.23/sysdeps/unix/sysv/linux/aarch64/configure.shareLocale~	2016-02-18 20:59:26.421203860 +0100
+++ glibc-2.23/sysdeps/unix/sysv/linux/aarch64/configure	2016-02-18 21:02:09.247980359 +0100
@@ -11,7 +11,7 @@ case "$prefix" in
   if test "$libdir" = '${exec_prefix}/lib'; then
     libdir='${exec_prefix}/lib64';
     # Locale data can be shared between 32-bit and 64-bit libraries.
-    libc_cv_complocaledir='${exec_prefix}/lib/locale'
+    libc_cv_complocaledir='${exec_prefix}/share/locale'
   fi
   ;;
 esac
--- glibc-2.23/sysdeps/unix/sysv/linux/x86_64/64/configure.shareLocale~	2016-02-18 20:59:26.421203860 +0100
+++ glibc-2.23/sysdeps/unix/sysv/linux/x86_64/64/configure	2016-02-18 21:03:10.050270476 +0100
@@ -9,7 +9,7 @@ case "$prefix" in
   if test "$libdir" = '${exec_prefix}/lib'; then
     libdir='${exec_prefix}/lib64';
     # Locale data can be shared between 32-bit and 64-bit libraries.
-    libc_cv_complocaledir='${exec_prefix}/lib/locale'
+    libc_cv_complocaledir='${exec_prefix}/share/locale'
   fi
   ;;
 esac
--- glibc-2.23/sysdeps/unix/sysv/linux/x86_64/x32/configure.shareLocale~	2016-02-18 20:59:26.422203865 +0100
+++ glibc-2.23/sysdeps/unix/sysv/linux/x86_64/x32/configure	2016-02-18 21:03:27.015351438 +0100
@@ -11,7 +11,7 @@ case "$prefix" in
   if test "$libdir" = '${exec_prefix}/lib'; then
     libdir='${exec_prefix}/libx32';
     # Locale data can be shared between 32-bit and 64-bit libraries.
-    libc_cv_complocaledir='${exec_prefix}/lib/locale'
+    libc_cv_complocaledir='${exec_prefix}/share/locale'
   fi
   ;;
 esac
