--- glibc-2.27/locale/programs/locarchive.c.0100~	2018-02-02 20:28:32.043266608 +0100
+++ glibc-2.27/locale/programs/locarchive.c	2018-02-02 20:29:20.894352169 +0100
@@ -1391,7 +1391,7 @@ add_locales_to_archive (size_t nlist, ch
 		    {
 		      char fullname[fnamelen + 2 * strlen (d->d_name) + 7];
 
-		      if (d_type == DT_UNKNOWN)
+		      if (d_type == DT_UNKNOWN || d_type == DT_LNK)
 			{
 			  strcpy (stpcpy (stpcpy (fullname, fname), "/"),
 				  d->d_name);
