--- libc/rt/Makefile.tsp~	2012-03-29 00:24:59.000000000 +0200
+++ libc/rt/Makefile	2012-12-01 22:44:45.732896177 +0100
@@ -78,3 +78,13 @@ $(tests:%=$(objpfx)%-bp): $(objpfx)librt
 endif
 
 tst-mqueue7-ARGS = -- $(local-built-program-cmd)
+
+# XXX avoid timing issues on fine-grained SMT systems
+ifeq (powerpc, $(base-machine))
+no-parallel-testing = yes
+endif
+ifneq ($(filter %tests,$(MAKECMDGOALS)),)
+ifeq ($(no-parallel-testing),yes)
+.NOTPARALLEL:
+endif
+endif
