--- glibc-2.23/crypt/crypt-entry.c.owlCrypt~	2016-02-18 18:54:00.000000000 +0100
+++ glibc-2.23/crypt/crypt-entry.c	2016-02-18 21:09:26.589068442 +0100
@@ -71,8 +71,8 @@ extern struct crypt_data _ufc_foobar;
  */
 
 char *
-__crypt_r (const char *key, const char *salt,
-	   struct crypt_data * __restrict data)
+__des_crypt_r (const char *key, const char *salt,
+	       struct crypt_data * __restrict data)
 {
   ufc_long res[4];
   char ktab[9];
@@ -143,6 +143,7 @@ __crypt_r (const char *key, const char *
   _ufc_output_conversion_r (res[0], res[1], salt, data);
   return data->crypt_3_buf;
 }
+#if 0
 weak_alias (__crypt_r, crypt_r)
 
 char *
@@ -166,6 +167,7 @@ crypt (const char *key, const char *salt
 
   return __crypt_r (key, salt, &_ufc_foobar);
 }
+#endif
 
 
 /*
--- glibc-2.23/crypt/Makefile.owlCrypt~	2016-02-18 18:54:00.000000000 +0100
+++ glibc-2.23/crypt/Makefile	2016-02-18 21:08:03.819673066 +0100
@@ -22,13 +22,13 @@ subdir	:= crypt
 
 include ../Makeconfig
 
-headers := crypt.h
+headers := crypt.h gnu-crypt.h ow-crypt.h
 
 extra-libs := libcrypt
 extra-libs-others := $(extra-libs)
 
 libcrypt-routines := crypt-entry md5-crypt sha256-crypt sha512-crypt crypt \
-		     crypt_util
+		     crypt_util crypt_blowfish x86 crypt_gensalt wrapper
 
 tests := cert md5c-test sha256c-test sha512c-test badsalttest
 
--- glibc-2.23/crypt/Versions.owlCrypt~	2016-02-18 18:54:00.000000000 +0100
+++ glibc-2.23/crypt/Versions	2016-02-18 21:08:03.819673066 +0100
@@ -1,5 +1,6 @@
 libcrypt {
   GLIBC_2.0 {
     crypt; crypt_r; encrypt; encrypt_r; fcrypt; setkey; setkey_r;
+    crypt_rn; crypt_ra; crypt_gensalt; crypt_gensalt_rn; crypt_gensalt_ra;
   }
 }
