--- libc/elf/rtld.c.rh737223~	2012-12-01 22:18:52.000000000 +0100
+++ libc/elf/rtld.c	2012-12-01 22:20:18.919961292 +0100
@@ -1405,7 +1405,10 @@ of this helper program; chances are you
 	      char *copy = malloc (len);
 	      if (copy == NULL)
 		_dl_fatal_printf ("out of memory\n");
-	      l->l_libname->name = l->l_name = memcpy (copy, dsoname, len);
+	      l->l_libname->name = memcpy (copy, dsoname, len);
+	      if (GLRO(dl_debug_mask))
+	        l->l_name = copy;
+
 	    }
 
 	  /* Add the vDSO to the object list.  */
