--- glibc-2.27/stdlib/Makefile.0118~	2018-02-02 20:31:07.402589690 +0100
+++ glibc-2.27/stdlib/Makefile	2018-02-02 20:31:53.077710331 +0100
@@ -190,9 +190,16 @@ test-canon-ARGS = --test-dir=${common-ob
 
 bug-fmtmsg1-ENV = SEV_LEVEL=foo,11,newsev
 
+isomac-includes = -I../include $(+sysdep-includes) $(sysincludes) -I..
+ifeq (powerpc:8, $(base-machine):$(sizeof-long-double))
+# XXX only necessary for bootstrapping...
+isomac-includes += -I../libio -I../stdlib -I../wcsmbs
+endif
+
 $(objpfx)isomac.out: $(objpfx)isomac
 	$(dir $<)$(notdir $<) '$(CC)'  \
 	'-I../include $(+sysdep-includes) $(sysincludes) -I..' > $@; \
+	'$(isomac-includes)' > $<.out \
 	$(evaluate-test)
 
 isomac-CFLAGS = -O
