--- glibc-2.18/crypt/crypt-entry.c.owlCrypt~	2013-08-11 00:52:55.000000000 +0200
+++ glibc-2.18/crypt/crypt-entry.c	2013-08-17 14:39:14.300517415 +0200
@@ -81,7 +81,7 @@ extern struct crypt_data _ufc_foobar;
  */
 
 char *
-__crypt_r (key, salt, data)
+__des_crypt_r (key, salt, data)
      const char *key;
      const char *salt;
      struct crypt_data * __restrict data;
@@ -155,6 +155,7 @@ __crypt_r (key, salt, data)
   _ufc_output_conversion_r (res[0], res[1], salt, data);
   return data->crypt_3_buf;
 }
+#if 0
 weak_alias (__crypt_r, crypt_r)
 
 char *
@@ -180,6 +181,7 @@ crypt (key, salt)
 
   return __crypt_r (key, salt, &_ufc_foobar);
 }
+#endif
 
 
 /*
--- glibc-2.18/crypt/Makefile.owlCrypt~	2013-08-17 14:38:07.574351607 +0200
+++ glibc-2.18/crypt/Makefile	2013-08-17 14:40:27.546601713 +0200
@@ -20,13 +20,13 @@
 #
 subdir	:= crypt
 
-headers := crypt.h
+headers := crypt.h gnu-crypt.h ow-crypt.h
 
 extra-libs := libcrypt
 extra-libs-others := $(extra-libs)
 
 libcrypt-routines := crypt-entry md5-crypt sha256-crypt sha512-crypt crypt \
-		     crypt_util
+		     crypt_util crypt_blowfish x86 crypt_gensalt wrapper
 
 tests := cert md5c-test sha256c-test sha512c-test badsalttest
 
--- glibc-2.18/crypt/Versions.owlCrypt~	2013-08-11 00:52:55.000000000 +0200
+++ glibc-2.18/crypt/Versions	2013-08-17 14:38:07.574351607 +0200
@@ -1,5 +1,6 @@
 libcrypt {
   GLIBC_2.0 {
     crypt; crypt_r; encrypt; encrypt_r; fcrypt; setkey; setkey_r;
+    crypt_rn; crypt_ra; crypt_gensalt; crypt_gensalt_rn; crypt_gensalt_ra;
   }
 }
