--- glibc-2.18/crypt/crypt-entry.c.relocateFcrypt~	2013-08-17 14:44:36.146493787 +0200
+++ glibc-2.18/crypt/crypt-entry.c	2013-08-17 14:44:42.994408176 +0200
@@ -182,20 +182,3 @@ crypt (key, salt)
   return __crypt_r (key, salt, &_ufc_foobar);
 }
 #endif
-
-
-/*
- * To make fcrypt users happy.
- * They don't need to call init_des.
- */
-#ifdef _LIBC
-weak_alias (crypt, fcrypt)
-#else
-char *
-__fcrypt (key, salt)
-     const char *key;
-     const char *salt;
-{
-  return crypt (key, salt);
-}
-#endif
--- glibc-2.18/crypt/wrapper.c.relocateFcrypt~	2013-08-17 14:44:36.143493824 +0200
+++ glibc-2.18/crypt/wrapper.c	2013-08-17 14:44:36.146493787 +0200
@@ -324,7 +324,6 @@ weak_alias(__crypt, crypt)
 weak_alias(__crypt_gensalt_rn, crypt_gensalt_rn)
 weak_alias(__crypt_gensalt_ra, crypt_gensalt_ra)
 weak_alias(__crypt_gensalt, crypt_gensalt)
-weak_alias(crypt, fcrypt)
 #endif
 
 #ifdef TEST
