--- glibc-2.18/stdlib/Makefile.ldbl~	2013-08-17 14:29:53.818524404 +0200
+++ glibc-2.18/stdlib/Makefile	2013-08-17 14:36:20.568689362 +0200
@@ -130,9 +130,15 @@ ifeq ($(run-built-tests),yes)
 tests: $(objpfx)tst-fmtmsg.out
 endif
 
+isomac-includes = -I../include $(+sysdep-includes) $(sysincludes) -I..
+ifeq (powerpc:8, $(base-machine):$(sizeof-long-double))
+# XXX only necessary for bootstrapping...
+isomac-includes += -I../libio -I../stdlib -I../wcsmbs
+endif
 $(objpfx)isomac.out: $(objpfx)isomac
 	$(dir $<)$(notdir $<) '$(CC)'  \
 	'-I../include $(+sysdep-includes) $(sysincludes) -I..' > $<.out
+	'$(isomac-includes)' > $<.out
 
 isomac-CFLAGS = -O
 $(objpfx)isomac: isomac.c
