--- glibc-2.25/crypt/crypt-entry.c.0122~	2017-02-05 16:28:43.000000000 +0100
+++ glibc-2.25/crypt/crypt-entry.c	2017-02-06 21:04:59.125974544 +0100
@@ -71,8 +71,8 @@ extern struct crypt_data _ufc_foobar;
  */
 
 char *
-__crypt_r (const char *key, const char *salt,
-	   struct crypt_data * __restrict data)
+__des_crypt_r (const char *key, const char *salt,
+	       struct crypt_data * __restrict data)
 {
   ufc_long res[4];
   char ktab[9];
@@ -152,6 +152,7 @@ __crypt_r (const char *key, const char *
 
   return data->crypt_3_buf;
 }
+#if 0
 weak_alias (__crypt_r, crypt_r)
 
 char *
@@ -175,6 +176,7 @@ crypt (const char *key, const char *salt
 
   return __crypt_r (key, salt, &_ufc_foobar);
 }
+#endif
 
 
 /*
--- glibc-2.25/crypt/Makefile.0122~	2017-02-05 16:28:43.000000000 +0100
+++ glibc-2.25/crypt/Makefile	2017-02-06 21:02:22.539522044 +0100
@@ -24,11 +24,11 @@ subdir	:= crypt
 
-headers := crypt.h
+headers := crypt.h gnu-crypt.h ow-crypt.h
 
 extra-libs := libcrypt
 extra-libs-others := $(extra-libs)
 
 libcrypt-routines := crypt-entry md5-crypt sha256-crypt sha512-crypt crypt \
-		     crypt_util
+		     crypt_util crypt_blowfish x86 crypt_gensalt wrapper
 
 tests := cert md5c-test sha256c-test sha512c-test badsalttest
 
--- glibc-2.25/crypt/Versions.0122~	2017-02-05 16:28:43.000000000 +0100
+++ glibc-2.25/crypt/Versions	2017-02-06 21:02:22.539522044 +0100
@@ -1,5 +1,6 @@
 libcrypt {
   GLIBC_2.0 {
     crypt; crypt_r; encrypt; encrypt_r; fcrypt; setkey; setkey_r;
+    crypt_rn; crypt_ra; crypt_gensalt; crypt_gensalt_rn; crypt_gensalt_ra;
   }
 }
