--- glibc-2.18/rt/Makefile.tsp~	2013-08-17 14:36:45.795373985 +0200
+++ glibc-2.18/rt/Makefile	2013-08-17 14:37:45.341629554 +0200
@@ -73,3 +73,13 @@ $(addprefix $(objpfx),$(tests)): $(objpf
 endif
 
 tst-mqueue7-ARGS = -- $(host-test-program-cmd)
+
+# XXX avoid timing issues on fine-grained SMT systems
+ifeq (powerpc, $(base-machine))
+no-parallel-testing = yes
+endif
+ifneq ($(filter %tests,$(MAKECMDGOALS)),)
+ifeq ($(no-parallel-testing),yes)
+.NOTPARALLEL:
+endif
+endif
