--- glibc-2.23/crypt/crypt-entry.c.relocateFcrypt~	2016-02-18 21:10:00.072228405 +0100
+++ glibc-2.23/crypt/crypt-entry.c	2016-02-18 21:10:58.203506144 +0100
@@ -168,18 +168,3 @@ crypt (const char *key, const char *salt
   return __crypt_r (key, salt, &_ufc_foobar);
 }
 #endif
-
-
-/*
- * To make fcrypt users happy.
- * They don't need to call init_des.
- */
-#ifdef _LIBC
-weak_alias (crypt, fcrypt)
-#else
-char *
-__fcrypt (const char *key, const char *salt)
-{
-  return crypt (key, salt);
-}
-#endif
--- glibc-2.23/crypt/wrapper.c.relocateFcrypt~	2016-02-18 21:10:00.071228400 +0100
+++ glibc-2.23/crypt/wrapper.c	2016-02-18 21:10:00.072228405 +0100
@@ -325,7 +325,6 @@ weak_alias(__crypt, crypt)
 weak_alias(__crypt_gensalt_rn, crypt_gensalt_rn)
 weak_alias(__crypt_gensalt_ra, crypt_gensalt_ra)
 weak_alias(__crypt_gensalt, crypt_gensalt)
-weak_alias(crypt, fcrypt)
 #endif
 
 #ifdef TEST
