diff -rup a/time/sys/time.h b/time/sys/time.h
--- a/time/sys/time.h	2012-01-01 05:16:32.000000000 -0700
+++ b/time/sys/time.h	2012-01-26 11:16:44.309112430 -0700
@@ -78,7 +78,7 @@ extern int gettimeofday (struct timeval
    This call is restricted to the super-user.  */
 extern int settimeofday (__const struct timeval *__tv,
 			 __const struct timezone *__tz)
-     __THROW __nonnull ((1));
+     __THROW;
 
 /* Adjust the current time of day by the amount in DELTA.
    If OLDDELTA is not NULL, it is filled in with the amount
