--- glibc-2.27/Makeconfig.0133~	2018-02-02 20:36:50.869704855 +0100
+++ glibc-2.27/Makeconfig	2018-02-02 20:37:32.598866637 +0100
@@ -360,6 +360,10 @@ else
 real-static-start-installed-name = $(static-start-installed-name)
 endif
 
+# glibc doesn't like gold and lld
+LDFLAGS.so += -fuse-ld=bfd
+LDFLAGS-rtld += -fuse-ld=bfd
+
 ifeq (yesyes,$(build-shared)$(have-z-combreloc))
 combreloc-LDFLAGS = -Wl,-z,combreloc
 LDFLAGS.so += $(combreloc-LDFLAGS)
