diff -up glibc-2.33/stdlib/canonicalize.c.omv~ glibc-2.33/stdlib/canonicalize.c
--- glibc-2.33/stdlib/canonicalize.c.omv~	2021-02-02 01:00:04.338272299 +0100
+++ glibc-2.33/stdlib/canonicalize.c	2021-02-02 00:56:17.245117135 +0100
@@ -413,7 +413,10 @@ error_nomem:
       return failed ? NULL : resolved;
     }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
   return scratch_buffer_dupfree (rname_buf, dest - rname);
+#pragma GCC diagnostic pop
 }
 
 /* Return the canonical absolute name of file NAME.  A canonical name
diff -up glibc-2.33/elf/cache.c.omv~ glibc-2.33/elf/cache.c
--- glibc-2.33/elf/cache.c.omv~	2021-02-02 01:12:23.403424474 +0100
+++ glibc-2.33/elf/cache.c	2021-02-02 01:18:28.015783023 +0100
@@ -756,6 +756,7 @@ save_cache (const char *cache_name)
       /* Align file position to 4.  */
       off64_t old_offset = lseek64 (fd, extension_offset, SEEK_SET);
       assert ((unsigned long long int) (extension_offset - old_offset) < 4);
+      (void)old_offset; /* Prevent unused variable warning with NDEBUG --> assert = nothing */
       write_extensions (fd, str_offset, extension_offset);
     }
 
