--- glibc-2.16.90-97bc38d7/misc/tst-error1.c.orig	2012-09-26 15:21:29.672179348 -0300
+++ glibc-2.16.90-97bc38d7/misc/tst-error1.c	2012-09-26 15:21:59.422180301 -0300
@@ -16,8 +16,8 @@ do_test (int argc, char *argv[])
   for (int i = 0; i < 1000; ++i)
     memcpy (&buf[i * (sizeof (str) - 1)], str, sizeof (str));
   error (0, 0, str);
-  error (0, 0, buf);
-  error (0, 0, buf);
+  error (0, 0, "%s", buf);
+  error (0, 0, "%s", buf);
   error (0, 0, str);
   return 0;
 }
--- glibc-2.16.90-97bc38d7/posix/regexbug1.c.orig	2012-09-26 15:23:13.175182590 -0300
+++ glibc-2.16.90-97bc38d7/posix/regexbug1.c	2012-09-26 15:23:32.198183235 -0300
@@ -18,7 +18,7 @@ main (void)
     {
       char buf[100];
       regerror (reerr, &re, buf, sizeof buf);
-      error (EXIT_FAILURE, 0, buf);
+      error (EXIT_FAILURE, 0, "%s", buf);
     }
 
   if (regexec (&re, "002", 2, ma, 0) != 0)
@@ -35,7 +35,7 @@ main (void)
     {
       char buf[100];
       regerror (reerr, &re, buf, sizeof buf);
-      error (EXIT_FAILURE, 0, buf);
+      error (EXIT_FAILURE, 0, "%s", buf);
     }
 
   if (regexec (&re, "002", 2, ma, 0) != 0)
--- glibc-2.16.90-97bc38d7/stdio-common/test-vfprintf.c.orig	2012-09-26 15:23:52.887183970 -0300
+++ glibc-2.16.90-97bc38d7/stdio-common/test-vfprintf.c	2012-09-26 15:24:13.870184603 -0300
@@ -92,7 +92,7 @@ main (void)
       fprintf (fp, "%s", large);
       fprintf (fp, "%.*s", 30000, large);
       large[20000] = '\0';
-      fprintf (fp, large);
+      fprintf (fp, "%s", large);
       fprintf (fp, "%-1.300000000s", "hello");
 
       if (fflush (fp) != 0 || ferror (fp) != 0 || fclose (fp) != 0)
