diff -up glibc-2.33/sysdeps/x86/configure.62~ glibc-2.33/sysdeps/x86/configure
--- glibc-2.33/sysdeps/x86/configure.62~	2021-03-11 00:59:44.937644191 +0100
+++ glibc-2.33/sysdeps/x86/configure	2021-03-11 01:04:04.372479500 +0100
@@ -136,7 +136,7 @@ if { ac_try='${CC-cc} $CFLAGS $CPPFLAGS
   test $ac_status = 0; }; }; then
   count=`LC_ALL=C $READELF -n conftest | grep NT_GNU_PROPERTY_TYPE_0 | wc -l`
   if test "$count" = 1; then
-    libc_cv_include_x86_isa_level=yes
+    libc_cv_include_x86_isa_level=no # workaround for vbox bug https://forums.gentoo.org/viewtopic-p-8568765.html?sid=563ab671df23b2a550273edc2dea30a2
     cat > conftest.c <<EOF
 EOF
     if { ac_try='${CC-cc} $CFLAGS $CPPFLAGS -fverbose-asm -S -o - conftest.c'
diff -up glibc-2.33/sysdeps/x86/configure.ac.62~ glibc-2.33/sysdeps/x86/configure.ac
--- glibc-2.33/sysdeps/x86/configure.ac.62~	2021-03-11 00:59:44.937644191 +0100
+++ glibc-2.33/sysdeps/x86/configure.ac	2021-03-11 01:03:29.158938280 +0100
@@ -103,7 +103,7 @@ libc_cv_have_x86_movbe=no
 if AC_TRY_COMMAND(${CC-cc} $CFLAGS $CPPFLAGS -nostartfiles -nostdlib -r -o conftest conftest1.S conftest2.S); then
   count=`LC_ALL=C $READELF -n conftest | grep NT_GNU_PROPERTY_TYPE_0 | wc -l`
   if test "$count" = 1; then
-    libc_cv_include_x86_isa_level=yes
+    libc_cv_include_x86_isa_level=no # workaround for vbox bug https://forums.gentoo.org/viewtopic-p-8568765.html?sid=563ab671df23b2a550273edc2dea30a2
     cat > conftest.c <<EOF
 EOF
     if AC_TRY_COMMAND(${CC-cc} $CFLAGS $CPPFLAGS -fverbose-asm -S -o - conftest.c) | grep -q "\-msahf"; then
