diff -pruN glibc-2.21-649-gae5eae7/rtkaio/Makefile glibc-2.21-649-gae5eae7.new/rtkaio/Makefile
--- glibc-2.21-649-gae5eae7/rtkaio/Makefile	2015-07-27 22:57:05.742601066 +0530
+++ glibc-2.21-649-gae5eae7.new/rtkaio/Makefile	2015-07-27 23:33:09.892874337 +0530
@@ -66,7 +66,9 @@ CFLAGS-kaio_librt-cancellation.c = -fasy
 
 LDFLAGS-rtkaio.so = -Wl,-soname=lib$(libprefix)rt.so$(librt.so-version) \
 		    -Wl,--enable-new-dtags,-z,nodelete
-CPPFLAGS-librtkaio += -DIS_IN_librt=1 -I$(..)rt
+# Resort to this ugliness of undefining and defining MODULE_NAME because
+# setting libof-<> to librt has many more side-effects that we want to avoid.
+CPPFLAGS-librtkaio += -I$(..)rt -UMODULE_NAME -DMODULE_NAME=librt
 
 rpath-dirs := $(patsubst rt,rtkaio,$(rpath-dirs))
 
diff -pruN glibc-2.21-649-gae5eae7/rtkaio/sysdeps/unix/sysv/linux/syscalls.list glibc-2.21-649-gae5eae7.new/rtkaio/sysdeps/unix/sysv/linux/syscalls.list
--- glibc-2.21-649-gae5eae7/rtkaio/sysdeps/unix/sysv/linux/syscalls.list	2015-07-27 22:47:23.073776396 +0530
+++ glibc-2.21-649-gae5eae7.new/rtkaio/sysdeps/unix/sysv/linux/syscalls.list	2015-07-27 23:33:09.892874337 +0530
@@ -1,5 +1,5 @@
 # File name		Caller	Syscall name	Args		Strong name		Weak names
 
-kaio_mq_timedsend	-	mq_timedsend	Ci:ipiip	__GI_mq_timedsend	mq_timedsend
-kaio_mq_timedreceive	-	mq_timedreceive	Ci:ipipp	__GI_mq_timedreceive	mq_timedreceive
-kaio_mq_setattr		-	mq_getsetattr	i:ipp		__GI_mq_setattr		mq_setattr
+kaio_mq_timedsend	-	mq_timedsend	Ci:ipiip	__mq_timedsend	mq_timedsend
+kaio_mq_timedreceive	-	mq_timedreceive	Ci:ipipp	__mq_timedreceive	mq_timedreceive
+kaio_mq_setattr	-	mq_getsetattr	i:ipp	mq_setattr
