--- glibc-2.16.90-54a41734/scripts/check-local-headers.sh.orig	2012-10-31 17:13:00.084315587 -0200
+++ glibc-2.16.90-54a41734/scripts/check-local-headers.sh	2012-10-31 17:13:44.515316810 -0200
@@ -31,7 +31,7 @@ exec ${AWK} -v includedir="$includedir"
 BEGIN {
   status = 0
   exclude = "^" includedir \
-    "/(.*-.*-.*/|)(asm[-/]|linux/|selinux/|gd|nss3/|sys/(capability|sdt(|-config))\\.h|libaudit\\.h)"
+    "/(.*-.*-.*/|)(asm[-/]|linux/|selinux/|multiarch-|gd|nss3/|sys/(capability|sdt(|-config))\\.h|libaudit\\.h)"
 }
 /^[^ ]/ && $1 ~ /.*:/ { obj = $1 }
 {
