diff -p -up glibc-2.9/Makeconfig.orig glibc-2.9/Makeconfig
--- glibc-2.9/Makeconfig.orig	2008-08-18 05:42:17.000000000 -0400
+++ glibc-2.9/Makeconfig	2009-01-15 09:44:31.000000000 -0500
@@ -226,7 +226,7 @@ inst_zonedir = $(install_root)$(zonedir)
 
 # Where to install the locale files.
 ifndef localedir
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 endif
 inst_localedir = $(install_root)$(localedir)
 
diff -p -up glibc-2.9/sysdeps/unix/sysv/linux/configure.in.orig glibc-2.9/sysdeps/unix/sysv/linux/configure.in
--- glibc-2.9/sysdeps/unix/sysv/linux/configure.in.orig	2008-04-11 17:13:38.000000000 -0400
+++ glibc-2.9/sysdeps/unix/sysv/linux/configure.in	2009-01-15 09:44:31.000000000 -0500
@@ -169,7 +169,7 @@ case "$prefix" in
     if test "$libdir" = '${exec_prefix}/lib'; then
       libdir='${exec_prefix}/lib64';
       # Locale data can be shared between 32bit and 64bit libraries
-      libc_cv_localedir='${exec_prefix}/lib/locale'
+      libc_cv_localedir='${datadir}/locale'
     fi
     ;;
   *)
diff -p -up glibc-2.9/sysdeps/unix/sysv/linux/configure.orig glibc-2.9/sysdeps/unix/sysv/linux/configure
--- glibc-2.9/sysdeps/unix/sysv/linux/configure.orig	2008-04-11 17:13:51.000000000 -0400
+++ glibc-2.9/sysdeps/unix/sysv/linux/configure	2009-01-15 09:44:31.000000000 -0500
@@ -381,7 +381,7 @@ case "$prefix" in
     if test "$libdir" = '${exec_prefix}/lib'; then
       libdir='${exec_prefix}/lib64';
       # Locale data can be shared between 32bit and 64bit libraries
-      libc_cv_localedir='${exec_prefix}/lib/locale'
+      libc_cv_localedir='${datadir}/locale'
     fi
     ;;
   *)
