diff -up glibc-2.30/nis/ypclnt.c.17~ glibc-2.30/nis/ypclnt.c
--- glibc-2.30/nis/ypclnt.c.17~	2019-08-02 13:54:49.077566424 +0200
+++ glibc-2.30/nis/ypclnt.c	2019-08-02 13:55:21.850394517 +0200
@@ -590,7 +590,7 @@ yp_master (const char *indomain, const c
 {
   ypreq_nokey req;
   ypresp_master resp;
-  enum clnt_stat result;
+  int result;
 
   if (indomain == NULL || indomain[0] == '\0'
       || inmap == NULL || inmap[0] == '\0')
