diff -up glibc-2.34/iconvdata/gconv-modules.19~ glibc-2.34/iconvdata/gconv-modules
--- glibc-2.34/iconvdata/gconv-modules.19~	2021-08-02 10:31:26.283941191 +0200
+++ glibc-2.34/iconvdata/gconv-modules	2021-08-02 10:32:27.975334292 +0200
@@ -113,3 +113,6 @@ module	INTERNAL		UTF-32BE//		UTF-32		1
 alias	UTF7//			UTF-7//
 module	UTF-7//			INTERNAL		UTF-7		1
 module	INTERNAL		UTF-7//			UTF-7		1
+
+alias	ISO-10646-UCS-2//	UNICODE//
+alias	ISO-10646-UCS-2//	ISO-10646/UTF8/
