diff -up glibc-2.43/string/strcoll_l.c.41~ glibc-2.43/string/strcoll_l.c
--- glibc-2.43/string/strcoll_l.c.41~	2026-01-23 21:54:00.000000000 +0100
+++ glibc-2.43/string/strcoll_l.c	2026-02-20 22:25:49.070372661 +0100
@@ -39,7 +39,11 @@
 #define CONCAT1(a,b) a##b
 
 #include "../locale/localeinfo.h"
+/* See Line 178+ */
+DIAG_PUSH_NEEDS_COMMENT;
+DIAG_IGNORE_Os_NEEDS_COMMENT (8, "-Wmaybe-uninitialized");
 #include WEIGHT_H
+DIAG_POP_NEEDS_COMMENT;
 
 /* Track status while looking for sequences in a string.  */
 typedef struct
@@ -291,7 +295,7 @@ STRCOLL (const STRING_TYPE *s1, const ST
 
   int result = 0, rule = 0;
 
-  /* With GCC 7 when compiling with -Os the compiler warns that
+  /* With GCC 7 and 8 when compiling with -Os the compiler warns that
      seq1.back_us and seq2.back_us might be used uninitialized.
      Sometimes this warning appears at locations in locale/weightwc.h
      where the actual use is, but on architectures other than x86_64,
@@ -299,7 +303,7 @@ STRCOLL (const STRING_TYPE *s1, const ST
      seq2.  This uninitialized use is impossible for the same reason
      as described in comments in locale/weightwc.h.  */
   DIAG_PUSH_NEEDS_COMMENT;
-  DIAG_IGNORE_Os_NEEDS_COMMENT_GCC (7, "-Wmaybe-uninitialized");
+  DIAG_IGNORE_Os_NEEDS_COMMENT_GCC (8, "-Wmaybe-uninitialized");
   coll_seq seq1, seq2;
   DIAG_POP_NEEDS_COMMENT;
   seq1.len = 0;
