diff -p -up glibc-2.10.1/crypt/Makefile.orig glibc-2.10.1/crypt/Makefile
--- glibc-2.10.1/crypt/Makefile.orig	2009-04-07 15:34:04.000000000 -0300
+++ glibc-2.10.1/crypt/Makefile	2009-05-27 00:02:34.000000000 -0300
@@ -21,13 +21,13 @@
 #
 subdir	:= crypt
 
-headers := crypt.h
+headers := crypt.h gnu-crypt.h ow-crypt.h
 
 extra-libs := libcrypt
 extra-libs-others := $(extra-libs)
 
 libcrypt-routines := crypt-entry md5-crypt sha256-crypt sha512-crypt crypt \
-		     crypt_util
+		     crypt_util crypt_blowfish x86 crypt_gensalt wrapper
 
 tests := cert md5c-test sha256c-test sha512c-test
 
diff -p -up glibc-2.10.1/crypt/Versions.orig glibc-2.10.1/crypt/Versions
--- glibc-2.10.1/crypt/Versions.orig	2000-03-03 21:47:30.000000000 -0300
+++ glibc-2.10.1/crypt/Versions	2009-05-26 23:58:49.000000000 -0300
@@ -1,5 +1,6 @@
 libcrypt {
   GLIBC_2.0 {
     crypt; crypt_r; encrypt; encrypt_r; fcrypt; setkey; setkey_r;
+    crypt_rn; crypt_ra; crypt_gensalt; crypt_gensalt_rn; crypt_gensalt_ra;
   }
 }
diff -p -up glibc-2.10.1/crypt/crypt-entry.c.orig glibc-2.10.1/crypt/crypt-entry.c
--- glibc-2.10.1/crypt/crypt-entry.c.orig	2007-09-19 17:37:15.000000000 -0300
+++ glibc-2.10.1/crypt/crypt-entry.c	2009-05-26 23:58:49.000000000 -0300
@@ -82,7 +82,7 @@ extern struct crypt_data _ufc_foobar;
  */
 
 char *
-__crypt_r (key, salt, data)
+__des_crypt_r (key, salt, data)
      const char *key;
      const char *salt;
      struct crypt_data * __restrict data;
@@ -137,6 +137,7 @@ __crypt_r (key, salt, data)
   _ufc_output_conversion_r (res[0], res[1], salt, data);
   return data->crypt_3_buf;
 }
+#if 0
 weak_alias (__crypt_r, crypt_r)
 
 char *
@@ -160,6 +161,7 @@ crypt (key, salt)
 
   return __crypt_r (key, salt, &_ufc_foobar);
 }
+#endif
 
 
 /*
