--- glibc-2.14-121-g5551a7b/streams/Makefile.fed_streams~	2011-07-27 15:13:29.702467500 +0000
+++ glibc-2.14-121-g5551a7b/streams/Makefile	2011-07-27 15:13:33.772466409 +0000
@@ -21,7 +21,7 @@
 #
 subdir	:= streams
 
-#headers		= stropts.h sys/stropts.h bits/stropts.h bits/xtitypes.h
+headers		= stropts.h sys/stropts.h bits/stropts.h bits/xtitypes.h
 routines	= isastream getmsg getpmsg putmsg putpmsg fattach fdetach
 
 include ../Rules
