--- glibc-2.16.90-54a41734/crypt/crypt-entry.c.orig	2012-10-31 17:17:00.738322781 -0200
+++ glibc-2.16.90-54a41734/crypt/crypt-entry.c	2012-10-31 17:17:54.401324787 -0200
@@ -81,7 +81,7 @@ extern struct crypt_data _ufc_foobar;
  */
 
 char *
-__crypt_r (key, salt, data)
+__des_crypt_r (key, salt, data)
      const char *key;
      const char *salt;
      struct crypt_data * __restrict data;
@@ -155,6 +155,7 @@ __crypt_r (key, salt, data)
   _ufc_output_conversion_r (res[0], res[1], salt, data);
   return data->crypt_3_buf;
 }
+#if 0
 weak_alias (__crypt_r, crypt_r)
 
 char *
@@ -180,7 +181,7 @@ crypt (key, salt)
 
   return __crypt_r (key, salt, &_ufc_foobar);
 }
-
+#endif
 
 /*
  * To make fcrypt users happy.
--- glibc-2.16.90-54a41734/crypt/Makefile.orig	2012-10-31 17:15:48.978320734 -0200
+++ glibc-2.16.90-54a41734/crypt/Makefile	2012-10-31 17:16:09.674320905 -0200
@@ -20,13 +20,13 @@
 #
 subdir	:= crypt
 
-headers := crypt.h
+headers := crypt.h gnu-crypt.h ow-crypt.h
 
 extra-libs := libcrypt
 extra-libs-others := $(extra-libs)
 
 libcrypt-routines := crypt-entry md5-crypt sha256-crypt sha512-crypt crypt \
-		     crypt_util
+		     crypt_util crypt_blowfish x86 crypt_gensalt wrapper
 
 tests := cert md5c-test sha256c-test sha512c-test badsalttest
 
--- glibc-2.16.90-54a41734/crypt/Versions.orig	2012-10-31 17:16:25.946322927 -0200
+++ glibc-2.16.90-54a41734/crypt/Versions	2012-10-31 17:16:43.186322523 -0200
@@ -1,5 +1,6 @@
 libcrypt {
   GLIBC_2.0 {
     crypt; crypt_r; encrypt; encrypt_r; fcrypt; setkey; setkey_r;
+    crypt_rn; crypt_ra; crypt_gensalt; crypt_gensalt_rn; crypt_gensalt_ra;
   }
 }
