diff -Naur glibc-2.29/localedata/Makefile glibc-2.29.tpg/localedata/Makefile
--- glibc-2.29/localedata/Makefile	2019-02-01 11:37:49.564873847 +0000
+++ glibc-2.29.tpg/localedata/Makefile	2019-02-01 11:38:20.352871703 +0000
@@ -413,6 +413,7 @@
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
 	$(LOCALEDEF) $$flags --alias-file=../intl/locale.alias \
+		     --no-archive \
 		     -i locales/$$input -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale \
 	&& echo ' done';
