--- glibc-2.15-a316c1f/scripts/check-local-headers.sh.orig	2012-02-11 17:00:16.708816468 -0200
+++ glibc-2.15-a316c1f/scripts/check-local-headers.sh	2012-02-11 17:00:35.235895337 -0200
@@ -29,7 +29,7 @@ exec ${AWK} -v includedir="$includedir"
 BEGIN {
   status = 0
   exclude = "^" includedir \
-    "/(asm[-/]|linux/|selinux/|gd|nss3/|sys/capability\\.h|libaudit\\.h|sys/sdt(-config)?\\.h))"
+    "/(asm[-/]|linux/|selinux/|multiarch-|gd|nss3/|sys/capability\\.h|libaudit\\.h|sys/sdt(-config)?\\.h))"
 }
 /^[^ ]/ && $1 ~ /.*:/ { obj = $1 }
 {
