diff -up glibc-2.28/sysdeps/unix/sysv/linux/riscv/flush-icache.c.omv~ glibc-2.28/sysdeps/unix/sysv/linux/riscv/flush-icache.c
--- glibc-2.28/sysdeps/unix/sysv/linux/riscv/flush-icache.c.omv~	2019-01-18 15:18:32.068258780 +0100
+++ glibc-2.28/sysdeps/unix/sysv/linux/riscv/flush-icache.c	2019-01-18 15:18:35.241304323 +0100
@@ -21,7 +21,6 @@
 #include <stdlib.h>
 #include <atomic.h>
 #include <sys/cachectl.h>
-#include <asm/syscalls.h>
 
 typedef int (*func_type) (void *, void *, unsigned long int);
 
