--- glibc-2.15-a316c1f/sysdeps/unix/sysv/linux/check_pf.c.orig	2012-02-11 15:25:40.066658360 -0200
+++ glibc-2.15-a316c1f/sysdeps/unix/sysv/linux/check_pf.c	2012-02-11 15:25:53.953718622 -0200
@@ -209,7 +209,8 @@ make_request (int fd, pid_t pid)
 		    }
 		  else
 		    {
-		      if (!IN6_IS_ADDR_LOOPBACK (address))
+		      if (!IN6_IS_ADDR_LOOPBACK (address) &&
+			  !IN6_IS_ADDR_LINKLOCAL (address))
 			seen_ipv6 = true;
 		    }
 		}
