--- glibc-2.16.90-97bc38d7/sysdeps/unix/sysv/linux/i386/sysdep.h.orig	2012-09-26 14:37:31.073094914 -0300
+++ glibc-2.16.90-97bc38d7/sysdeps/unix/sysv/linux/i386/sysdep.h	2012-09-26 14:37:42.928095292 -0300
@@ -27,6 +27,10 @@
 #include <dl-sysdep.h>
 #include <tls.h>
 
+#if defined __i686 && defined __ASSEMBLER__
+#undef __i686
+#define __i686 __i686
+#endif
 
 /* For Linux we can use the system call table in the header file
 	/usr/include/asm/unistd.h
