diff -up glibc-2.43/malloc/malloc.c.23~ glibc-2.43/malloc/malloc.c
--- glibc-2.43/malloc/malloc.c.23~	2026-01-23 21:54:00.000000000 +0100
+++ glibc-2.43/malloc/malloc.c	2026-02-20 22:23:00.244168292 +0100
@@ -2270,6 +2270,9 @@ sysmalloc_mmap (INTERNAL_SIZE_T nb, size
 
   __set_vma_name (mm, size, " glibc: malloc");
 
+  if (size > 8 * 1024 * 1024)
+    __madvise (mm, size , MADV_HUGEPAGE);
+
   mchunkptr p = mmap_set_chunk ((uintptr_t)mm, size, padding, extra_flags != 0);
 
   /* update statistics */
