diff -up glibc-2.35/malloc/malloc.c.23~ glibc-2.35/malloc/malloc.c
--- glibc-2.35/malloc/malloc.c.23~	2022-02-03 11:12:21.887241710 +0100
+++ glibc-2.35/malloc/malloc.c	2022-02-03 11:13:26.221492043 +0100
@@ -2449,6 +2449,9 @@ sysmalloc_mmap (INTERNAL_SIZE_T nb, size
     madvise_thp (mm, size);
 #endif
 
+  if (size > 8 * 1024 * 1024)
+    __madvise (mm, size , MADV_HUGEPAGE);
+
   /*
     The offset to the start of the mmapped region is stored in the prev_size
     field of the chunk.  This allows us to adjust returned start address to
