diff -up glibc-2.42/elf/dl-map-segments.h.17~ glibc-2.42/elf/dl-map-segments.h
--- glibc-2.42/elf/dl-map-segments.h.17~	2025-07-28 20:22:55.000000000 +0200
+++ glibc-2.42/elf/dl-map-segments.h	2025-07-30 00:15:00.568100672 +0200
@@ -191,6 +191,7 @@ _dl_map_segments (struct link_map *l, in
                               -1, 0);
               if (__glibc_unlikely (mapat == MAP_FAILED))
                 return DL_MAP_SEGMENTS_ERROR_MAP_ZERO_FILL;
+	      __madvise((caddr_t) zeropage, zeroend - zeropage, MADV_MERGEABLE);
               if (__is_decorate_maps_enabled ())
                 {
                   if (l->l_name != NULL && *l->l_name != '\0')
