diff -Naur glibc-2.26/crypt/Makefile glibc-2.26.tpg/crypt/Makefile
--- glibc-2.26/crypt/Makefile	2017-08-02 12:57:16.000000000 +0000
+++ glibc-2.26.tpg/crypt/Makefile	2018-01-03 22:26:46.071151667 +0000
@@ -22,6 +22,8 @@
 
 include ../Makeconfig
 
+CFLAGS-.o += -flto -ffat-lto-objects -fno-stack-protector
+
 headers := crypt.h
 
 extra-libs := libcrypt
diff -Naur glibc-2.26/math/Makefile glibc-2.26.tpg/math/Makefile
--- glibc-2.26/math/Makefile	2018-01-03 22:25:33.065056000 +0000
+++ glibc-2.26.tpg/math/Makefile	2018-01-03 22:27:03.487174438 +0000
@@ -21,6 +21,8 @@
 
 include ../Makeconfig
 
+CFLAGS-.o += -fno-stack-protector
+
 # Installed header files.
 headers		:= math.h bits/mathcalls.h bits/mathinline.h \
 		   fpu_control.h complex.h bits/cmathcalls.h fenv.h \
