diff -Naur glibc-2.29/crypt/Makefile glibc-2.29.tpg/crypt/Makefile
--- glibc-2.29/crypt/Makefile	2019-01-31 16:45:36.000000000 +0000
+++ glibc-2.29.tpg/crypt/Makefile	2019-04-17 17:32:03.818000000 +0000
@@ -22,6 +22,8 @@
 
 include ../Makeconfig
 
+CFLAGS-.o += -flto -ffat-lto-objects -fno-stack-protector
+
 headers := crypt.h
 
 extra-libs := libcrypt
diff -Naur glibc-2.29/math/Makefile glibc-2.29.tpg/math/Makefile
--- glibc-2.29/math/Makefile	2019-01-31 16:45:36.000000000 +0000
+++ glibc-2.29.tpg/math/Makefile	2019-04-17 17:31:44.311000000 +0000
@@ -21,6 +21,8 @@
 
 include ../Makeconfig
 
+CFLAGS-.o += -fno-stack-protector
+
 # Installed header files.
 headers		:= math.h bits/mathcalls.h bits/mathinline.h \
 		   fpu_control.h complex.h bits/cmathcalls.h fenv.h \
