--- libc/nptl/nptl-init.c.bero~	2012-12-01 23:55:26.722754391 +0100
+++ libc/nptl/nptl-init.c	2012-12-01 23:55:42.725950817 +0100
@@ -414,7 +414,7 @@ __pthread_initialize_minimal_internal (v
   /* Determine the default allowed stack size.  This is the size used
      in case the user does not specify one.  */
   struct rlimit limit;
-  if (__getrlimit (RLIMIT_STACK, &limit) != 0
+  if (getrlimit (RLIMIT_STACK, &limit) != 0
       || limit.rlim_cur == RLIM_INFINITY)
     /* The system limit is not usable.  Use an architecture-specific
        default.  */
