diff -up glibc-2.35/nss/XXX-lookup.c.185~ glibc-2.35/nss/XXX-lookup.c
--- glibc-2.35/nss/XXX-lookup.c.185~	2022-06-24 23:19:42.023870561 +0200
+++ glibc-2.35/nss/XXX-lookup.c	2022-06-24 23:20:35.490963277 +0200
@@ -53,6 +53,9 @@ DB_LOOKUP_FCT (nss_action_list *ni, cons
   if (! __nss_database_get (DATABASE_NAME_ID, &DATABASE_NAME_SYMBOL))
     return -1;
 
+  if (!DATABASE_NAME_SYMBOL)
+    return -1;
+
   *ni = DATABASE_NAME_SYMBOL;
 
   /* We want to know about it if we've somehow got a NULL action list;
