diff -up glibc-2.43/sysdeps/x86/atomic-machine.h.26~ glibc-2.43/sysdeps/x86/atomic-machine.h
--- glibc-2.43/sysdeps/x86/atomic-machine.h.26~	2026-01-23 21:54:00.000000000 +0100
+++ glibc-2.43/sysdeps/x86/atomic-machine.h	2026-02-20 22:23:48.662773148 +0100
@@ -19,6 +19,6 @@
 #ifndef _X86_ATOMIC_MACHINE_H
 #define _X86_ATOMIC_MACHINE_H 1
 
-#define atomic_spin_nop() __asm ("pause")
+#define atomic_spin_nop() __asm ("lfence")
 
 #endif /* atomic-machine.h */
