--- a/Makeconfig	2026-07-24 15:16:45.000000000 +0200
+++ b/Makeconfig	2026-07-25 21:17:13.323444235 +0200
@@ -380,6 +380,15 @@
 LDFLAGS.so += $(relro-LDFLAGS)
 LDFLAGS-rtld += $(relro-LDFLAGS)
 
+ifeq (yes,$(have-hash-style))
+# For the time being we unconditionally use 'both'.  At some time we
+# should declare statically linked code as 'out of luck' and compile
+# with --hash-style=gnu only.
+hashstyle-LDFLAGS = -Wl,--hash-style=both
+LDFLAGS.so += $(hashstyle-LDFLAGS)
+LDFLAGS-rtld += $(hashstyle-LDFLAGS)
+endif
+
 # Linker options to enable and disable DT_RELR.
 ifeq ($(have-dt-relr),yes)
 dt-relr-ldflag = -Wl,-z,pack-relative-relocs
--- a/config.make.in	2026-07-24 15:16:45.000000000 +0200
+++ b/config.make.in	2026-07-25 21:17:13.323498055 +0200
@@ -70,6 +70,7 @@
 have-libaudit = @have_libaudit@
 have-libcap = @have_libcap@
 bind-now = @bindnow@
+have-hash-style = @libc_cv_hashstyle@
 have-cxx-thread_local = @libc_cv_cxx_thread_local@
 have-loop-to-function = @libc_cv_cc_loop_to_function@
 have-textrel_ifunc = @libc_cv_textrel_ifunc@
--- a/configure.ac	2026-07-24 15:16:45.000000000 +0200
+++ b/configure.ac	2026-07-25 21:17:13.323518136 +0200
@@ -1343,6 +1343,22 @@
 
 AC_SUBST(libc_cv_fpie)
 
+AC_CACHE_CHECK(for --hash-style option,
+	       libc_cv_hashstyle, [dnl
+cat > conftest.c <<EOF
+int _start (void) { return 42; }
+EOF
+if AC_TRY_COMMAND([${CC-cc} $CFLAGS $CPPFLAGS $LDFLAGS $no_ssp
+			    -fPIC -shared -o conftest.so conftest.c
+			    -Wl,--hash-style=both -nostdlib 1>&AS_MESSAGE_LOG_FD])
+then
+  libc_cv_hashstyle=yes
+else
+  libc_cv_hashstyle=no
+fi
+rm -f conftest*])
+AC_SUBST(libc_cv_hashstyle)
+
 AC_CACHE_CHECK(for GLOB_DAT reloc,
 	       libc_cv_has_glob_dat, [dnl
 cat > conftest.c <<EOF
