diff -up glibc-2.35/nptl/pthread_mutex_lock.c.27~ glibc-2.35/nptl/pthread_mutex_lock.c
--- glibc-2.35/nptl/pthread_mutex_lock.c.27~	2022-02-03 11:13:51.081588372 +0100
+++ glibc-2.35/nptl/pthread_mutex_lock.c	2022-02-03 11:16:15.380143680 +0100
@@ -145,7 +145,13 @@ PTHREAD_MUTEX_LOCK (pthread_mutex_t *mut
 		  LLL_MUTEX_LOCK (mutex);
 		  break;
 		}
-	      atomic_spin_nop ();
+		/* MO read while spinning */
+		do
+		  {
+		    atomic_spin_nop ();
+		  }
+		while (atomic_load_relaxed (&mutex->__data.__lock) != 0 &&
+			++cnt < max_cnt);
 	    }
 	  while (LLL_MUTEX_READ_LOCK (mutex) != 0
 		 || LLL_MUTEX_TRYLOCK (mutex) != 0);
